crescent/plugin.py

```diff
@@ -9,27 +10,30 @@
 if TYPE_CHECKING:
     from typing import Any, Sequence, TypeVar
 
-    from crescent.typedefs import HookCallbackT
-
-    from .bot import Bot
+    from crescent.bot import Bot
+    from crescent.typedefs import HookCallbackT, PluginCallbackT
 
     T = TypeVar("T", bound="MetaStruct[Any, Any]")
 
 
 __all__: Sequence[str] = ("PluginManager", "Plugin")
 
+_LOG = getLogger(__name__)
+
 
 class PluginManager:
     def __init__(self, bot: Bot) -> None:
         self.plugins: Dict[str, Plugin] = {}
         self._bot = bot
 
     def add_plugin(self, plugin: Plugin, force: bool = False) -> None:
-        if plugin.name in self.plugins:
-            if not force:
-                raise ValueError(f"Plugin name {plugin.name} already exists.")
-        self.plugins[plugin.name] = plugin
-        plugin._setup(self._bot)
+        _LOG.warning("`add_plugin` is deprecated and will be removed in a future release.")
+
+        self._add_plugin(plugin, refresh=force)
+
+    def unload(self, name: str) -> None:
```

Comment:
Perhaps allow unload to accept a path, rather than only a plugin name?
```py
def unload(self, name: str | None, path: str | None) -> None:
    if bool(name) is bool(path):
        raise ValueError("You must specify a plugin name or path (not both).")
    elif name:
        self.plugins.pop(name)._unload()
    else:
        Plugin._from_module(path)._unload()
```

Reply:
Probably deprecate name in this PR tbh.